internal/pkg/pkg.go

```diff
@@ -123,20 +128,15 @@ type Pkg struct {
 
 // New returns a pkg given an absolute or relative OS-defined path.
 // Use ReadKptfile or ReadPipeline on the return value to read meta resources from filesystem.
-func New(path string) (*Pkg, error) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		return nil, err
-	}
+func New(fs filesys.FileSystem, path string) (*Pkg, error) {
 	var absPath string
 	if filepath.IsAbs(path) {
```

Comment:
This can be done, but since this is feature branch, we can just pull the plug and avoid the transition phase. I have updated the callers in the PR already.